index.channel.ts

```diff
@@ -0,0 +1,441 @@
+    });
+  }
+
+  handle(req: Request | SocketRequest, res: Response | SocketResponse) {
+    throw new Error('Discord Channel Handler is not using Webhooks currently.');
+  }
+
+  /**
+   * Format any type of message
+   *
+   * @param envelope - The message standard envelope
+   * @param options - The block options related to the message
+   *
+   * @returns A template filled with its payload
+   */
+  _formatMessage(envelope: StdOutgoingEnvelope, options: BlockOptions): any {
+    switch (envelope.format) {
+      case OutgoingMessageFormat.attachment:
+        return this._attachmentFormat(envelope.message, options);
+      case OutgoingMessageFormat.buttons:
+        return this._buttonsFormat(envelope.message, options);
+      case OutgoingMessageFormat.carousel:
+        return this._carouselFormat(envelope.message, options);
+      case OutgoingMessageFormat.list:
+        return this._listFormat(envelope.message, options);
+      case OutgoingMessageFormat.quickReplies:
+        return this._quickRepliesFormat(envelope.message, options);
+      case OutgoingMessageFormat.text:
+        return this._textFormat(envelope.message, options);
+
+      default:
+        throw new Error('Unknown message format');
+    }
+  }
+
+  async sendMessage(
+    event: DiscordEventWrapper,
+    envelope: StdOutgoingEnvelope,
+    options: any,
+    context: any,
+  ): Promise<{ mid: string }> {
+    try {
+      this.logger.log('Discord Channel Handler: Sending message ...');
+      debugger;
+      const client = this.discordBotService.getClient();
+      const handler: DiscordChannelHandler = this;
+      const originalEvent:
+        | ButtonInteraction
+        | ChatInputCommandInteraction
+        | Message = event.getOriginalEvent();
+
+      const payload = await handler._formatMessage(envelope, options);
+      const message = event.getOriginalMessageContent();
+
+      if (originalEvent.channel.type === ChannelType.GuildText) {
+        // Create embed to show user's message
+        const embed = new EmbedBuilder()
+          .setColor(0x6cc853)
+          .addFields(
+            {
+              name: `${event.getSender().first_name}:`,
+              value: message,
+            },
+            {
+              name: 'Hexabot:',
+              value: payload?.content || '\u200B',
+            },
+          )
+          .setAuthor({
+            name: '\u200B',
+            iconURL: event.getProfile().avatarUrl,
+          });
+        if (payload.files) {
+          embed.setImage('attachment://' + payload.files[0].name);
+        }
+
+        const res = await (
+          originalEvent as ButtonInteraction | ChatInputCommandInteraction
+        ).followUp({
+          embeds: [embed],
+          components: payload.components,
+          files: payload.files,
+        });
+        return { mid: res.id };
+      }
+
+      if (originalEvent.channel.type === ChannelType.DM) {
+        const channel = await client.channels.fetch(
```

Comment:
```suggestion
        const slackChannel = await client.channels.fetch(
```
Can we rename any slack channel to avoid any confusion. ?